internal/declarative/v2/reconciler_test.go

```diff
@@ -27,14 +29,32 @@ func TestPruneKymaSystem(t *testing.T) {
 			service,
 			kymaNs,
 			deployment,
+			crd,
 		}
 
-		result := pruneKymaSystem(infos)
+		result := pruneResource(infos, "Namespace", namespaceNotBeRemoved)
 
-		require.Len(t, result, 3)
+		require.Len(t, result, 4)
 		require.NotContains(t, result, kymaNs)
 	})
 
+	t.Run("prune a crd", func(t *testing.T) {
+		t.Parallel()
+
+		infos := []*resource.Info{
+			kubeNs,
+			service,
+			kymaNs,
+			deployment,
+			crd,
+		}
+
+		result := pruneResource(infos, "CustomResourceDefinition", "some-crd")
```

Comment:
please replace those two `"some-crd"` with a variable so that others know why you can prune this crd